wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+/// A property wrapper which allows easily implementing Copy-on-Write for value types.
+@propertyWrapper
+public struct CopyOnWrite<Value> {
```

Comment:
Noting: This is a copy of the market implementation of copy on write. It's largely worked, so seems fine to move here.